src/components/Link/Item.js

```diff
@@ -59,20 +61,74 @@ export default class LinkItem extends React.Component {
   };
 
   componentDidMount() {
+    const { open } = this.state;
+    const { disableHoverEvents } = this.props;
+
     if ("ontouchstart" in document.documentElement) {
       this.setState({ touchEventsExist: true }); // eslint-disable-line
     }
+
+    if (open && disableHoverEvents && !this.hasListener) this.attachListener();
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const { open: prevOpen } = prevState;
+    const { open } = this.state;
+    const { disableHoverEvents } = this.props;
+
+    if (!prevOpen && open && disableHoverEvents && !this.hasListener) {
+      this.attachListener();
+    } else if (prevOpen && !open && this.hasListener) {
+      this.removeListener();
+    }
   }
 
+  componentWillUnmount() {
+    if (this.hasListener) this.removeListener();
+  }
+
+  hasListener = false;
+
+  ref = this.props.disableHoverEvents ? createRef() : undefined;
+
   open = () => this.hasOther && this.setState(() => ({ open: true }));
 
   close = () => this.hasOther && this.setState(() => ({ open: false }));
 
   toggle = () =>
     this.hasOther && this.setState(({ open }) => ({ open: !open }));
 
+  handleOutsideClick = e => {
+    if (this.ref && this.ref.current && !this.ref.current.contains(e.target)) {
+      this.close();
+    }
+  };
```

Comment:
Do we have test coverage on this?

Reply:
I think I forgot this case. I will add a test. Thanks!